crates/polars-utils/src/mem.rs

```diff
@@ -23,3 +23,54 @@ pub unsafe fn prefetch_l2(ptr: *const u8) {
         unsafe { _prefetch(ptr as *const _, _PREFETCH_READ, _PREFETCH_LOCALITY2) };
     }
 }
+
+/// Attempt to prefetch the memory in the slice.
+pub fn prefetch_l2(slice: &[u8]) {
+    if slice.is_empty() {
+        return;
+    }
+
+    // @TODO: We can play a bit more with this prefetching. Maybe introduce a maximum number of
+    // prefetches as to not overwhelm the processor. The linear prefetcher should pick it up
+    // at a certain point.
+
+    const PAGE_SIZE: usize = 4096;
+
+    for i in (0..slice.len()).step_by(PAGE_SIZE) {
+        unsafe { prefetch_l2_impl(slice[i..].as_ptr()) };
+    }
+
+    unsafe { prefetch_l2_impl(slice[slice.len() - 1..].as_ptr()) }
+}
```

Comment:
Factories this logic out from `MemSlice::prefetch` so that we can use it without constructing a `MemSlice`